packages/babel-generator/test/fixtures/types/MatchExpression/expected.js

```diff
@@ -0,0 +1,3 @@
+let a = match(b) {
+  "3": "4"
+};
```

Comment:
I think there is an issue here, the expected output wasn't transpiled.

And I noticed that the transpiled version is incorrect:
```js
var a = function () {
  {
    if (b === "3") return "4";else throw new Error("No patterns are matched");
  }
}();
```

It seems that the `b` binding is missing.

Reply:
@xtuc I don't understand. I suppose `babel-generator`  just generate code from ASTs, it should not transform the ASTs.

And I think the transpiled version you pinned is right, what should `b` bind to?

Reply:
@vincentdchan It would rather be: 

```javascript
var a = function (b) {
  {
    if (b === "3") return "4";else throw new Error("No patterns are matched");
  }
}();
```
? Otherwise, how do we expect the `(b === 3)` to run ?

Reply:
`b` is defined somewhere else. e.g.
```js
let b = "3";
let a = match(b) {
  "3": "4"
}
```

Reply:
@vincentdchan yes sorry this is indeed the generator. 

Nicolo is right, the fact that the variable isn't defined here confused me. 
What is the best: pass it as arg or as closure? Probably not the same from the memory point of view.

Reply:
@titouancreach I suppose you want to write:
```js
var a = function (b) {
  {
    if (b === "3") return "4";else throw new Error("No patterns are matched");
  }
}(b); // <-- parameter
```

@xtuc I think the differences between arg and closure can be ignored. The function is disposable. After it's called, the closure will be released by GC. Actually, the script engine decides those things, I focus on the correctness of syntax.

Reply:
@vincentdchan Yes exactly

Reply:
Ok sounds good